test/acceptance/authz/roles_test.go

```diff
@@ -723,3 +723,204 @@ func TestAuthzRoleRemoveToEmptyAndAddPermission(t *testing.T) {
+		require.Equal(t, 1, len(roles))
+		require.Equal(t, limitedRole, *roles[0].Name)
+	})
+
+	t.Run("limited user can create role with equal permissions", func(t *testing.T) {
+		_, err = helper.Client(t).Authz.CreateRole(
+			authz.NewCreateRoleParams().WithBody(&models.Role{
+				Name: &newRole,
+				Permissions: []*models.Permission{
+					{
+						Action: String(authorization.CreateCollections),
+						Collections: &models.PermissionCollections{
+							Collection: String("Collection1"),
+						},
+					},
+				},
+			}),
+			limitedAuth,
+		)
+		require.NoError(t, err)
+	})
+
+	t.Run("limited user cannot create role with broader permissions", func(t *testing.T) {
+		_, err = helper.Client(t).Authz.CreateRole(
+			authz.NewCreateRoleParams().WithBody(&models.Role{
+				Name: &broaderRole,
+				Permissions: []*models.Permission{
+					{
+						Action: String(authorization.CreateCollections),
+						Collections: &models.PermissionCollections{
+							Collection: String("*"),
+						},
+					},
+				},
+			}),
+			limitedAuth,
+		)
+		require.Error(t, err)
+		var parsed *authz.CreateRoleForbidden
+		require.True(t, errors.As(err, &parsed))
+	})
+
+	t.Run("limited user can update role within their scope", func(t *testing.T) {
+		_, err = helper.Client(t).Authz.AddPermissions(
+			authz.NewAddPermissionsParams().WithID(newRole).WithBody(authz.AddPermissionsBody{
+				Permissions: []*models.Permission{
+					{
+						Action: String(authorization.UpdateCollections),
+						Collections: &models.PermissionCollections{
+							Collection: String("Collection1"),
+						},
+					},
+				},
+			}),
+			limitedAuth,
+		)
+		require.NoError(t, err)
+	})
+
+	t.Run("limited user cannot update role beyond their scope(AddPermission)", func(t *testing.T) {
+		_, err = helper.Client(t).Authz.AddPermissions(
+			authz.NewAddPermissionsParams().WithID(newRole).WithBody(authz.AddPermissionsBody{
+				Permissions: []*models.Permission{
+					{
+						Action: String(authorization.CreateCollections),
+						Collections: &models.PermissionCollections{
+							Collection: String("Collection2"),
+						},
+					},
+				},
+			}),
+			limitedAuth,
+		)
+		require.Error(t, err)
+		var parsed *authz.AddPermissionsForbidden
+		require.True(t, errors.As(err, &parsed))
+	})
+
+	t.Run("limited user cannot update role beyond their scope(RemovePermission)", func(t *testing.T) {
+		_, err = helper.Client(t).Authz.RemovePermissions(
+			authz.NewRemovePermissionsParams().WithID(newRole).WithBody(authz.RemovePermissionsBody{
+				Permissions: []*models.Permission{
+					{
+						Action: String(authorization.UpdateCollections),
+						Collections: &models.PermissionCollections{
+							Collection: String("Collection2"),
+						},
+					},
+				},
+			}),
+			limitedAuth,
+		)
+		require.Error(t, err)
+		var parsed *authz.RemovePermissionsForbidden
+		require.True(t, errors.As(err, &parsed))
+	})
+
+	t.Run("limited user can not remove permissions from role", func(t *testing.T) {
```

Comment:
this is almost identical to the test above, did you want to test deleting a role with wider permissions?

Reply:
the name was just wrong, this the user shall be able to remove the above the user shall not be able to